regolith/helpers/l_membershelper.py

```diff
@@ -67,15 +67,28 @@ def sout(self):
         bad_stati = ["finished", "cancelled", "paused", "back_burner"]
         people = []
         for person in self.gtx["people"]:
-            if rc.current and not person.get('active'):
-                continue
-            people.append(person)
-        #people.sort()
+            if rc.current:
+                if not person.get('active'):
+                    continue
+                people.append(person)
+            elif rc.former:
+                if person.get('active'):
+                    continue
+                people.append(person)
+            else:
+                people.append(person)
+
         for i in people:
             if rc.verbose:
                 print("{}, {} | group_id: {}".format(i.get('name'), i.get('position'), i.get('_id')))
                 print("    orcid: {} | github_id: {}".format(i.get('orcid_id'), i.get('github_id')))
+                for employment in i.get('employment'):
```

Comment:
Assuming the appointments are time sorted in the database is not safe.   This should fix it though:
```
positions = [emp for emp in i.get('employment')].sort(key=lambda x: get_dates(x)["end_date"])
for position in positions:
...
```
if not already you can import ``get_dates`` from ``regolith.dates``

Write a test with the appointments misordered to test this (i.e., update the exemplar schema this way)

Reply:
@sbillinge When I try the lambda option, the sort was giving errors. I know it is better for optimizing but I could not come up with how to solve this so wanted to ask how to approach it to be able to use the sort:
TypeError: 'NoneType' object is not iterable
TypeError: '<' not supported between instances of 'datetime.date' and 'NoneType'

Reply:
@sbillinge The following block of code is producing empty list and Zach and I can't figure it out why, could you guide us through?
                not_current_positions = [emp for emp in i.get('employment') if not is_current(emp)].sort(key=lambda x: 
                                                             get_dates(x)["end_date"])
                if not_current_positions is None:
                    not_current_positions = []
                current_positions = [emp for emp in i.get('employment') if is_current(emp)].sort(
                                                      key=lambda x: get_dates(x)["begin_date"])
                if current_positions is None:
                    current_positions = []
                positions = not_current_positions + current_positions

Reply:
I would have to fiddle around with it.  We could try and do it on a call.